### PR TITLE
feat: add fast-forwarding

### PR DESCRIPTION
**Pull request by ChaoticTempest**
Adds the `worker.fast_forward(x)` specific API for simulating producing `x` number of blocks on the sandbox node. Also adds a simple example for fast forwarding and grabbing time specific data related to how many blocks have been produced before and after a `fast_forward` request.

Mainly opening this one up for comments so we can review this one sooner than later, and will point to the specific nearcore version once https://github.com/near/nearcore/pull/6211 merges

### CODE REVIEW

File: workspaces/Cargo.toml [hunk 1]
```diff
@@ -26,15 +26,15 @@ tokio-retry = "0.3"
...
-near-crypto = "0.5"
-near-primitives = "0.5"
-near-jsonrpc-primitives = "0.5"
-near-jsonrpc-client = { version = "0.1", features = ["sandbox"] }
-near-sandbox-utils = "0.1.1"
+near-account-id = "0.12"
+near-crypto = "0.12"
+near-primitives = "0.12"
+near-jsonrpc-primitives = "0.12"
+near-jsonrpc-client = { git = "https://github.com/near/near-jsonrpc-client-rs", features = ["sandbox"] }
```
Comment (miraclx) [inline workspaces/Cargo.toml:33]: `near-jsonrpc-client` `v0.3.0` has been released. https://github.com/near/near-jsonrpc-client-rs/releases/tag/v0.3.0


```suggestion
near-jsonrpc-client = { version = "0.3", features = ["sandbox"] }
```

**Review by miraclx — CHANGES_REQUESTED** [review]

File: workspaces/src/network/mod.rs [hunk 2]
```diff
@@ -151,6 +154,18 @@ where
     ) -> ImportContractTransaction<'a, 'b> {
         ImportContractTransaction::new(id.to_owned(), worker.client(), self.client())
     }
+
+    async fn fast_forward(&self, delta_height: u64) -> anyhow::Result<()> {
+        // NOTE: RpcSandboxFastForwardResponse is an empty struct with no fields, so don't do anything with it:
+        let _forward_resp = self
```
Comment (austinabell) [inline workspaces/src/network/mod.rs:160]:
```suggestion
        self
```

If you're not checking the response serialization is correct, probably no need to bind this to a variable

File: workspaces/src/worker/impls.rs [hunk 3]
```diff
@@ -77,6 +77,10 @@ where
     ) -> ImportContractTransaction<'a, 'b> {
         self.workspace.import_contract(id, worker)
     }
+
+    async fn fast_forward(&self, delta_height: u64) -> anyhow::Result<()> {
```
Comment (austinabell) [inline workspaces/src/worker/impls.rs:81]: noting that adding this function to `StatePatcher` blurs that naming meaning. These things seem sandbox specific so not sure why the trait is needed yet

**Review by austinabell — APPROVED** [review]

File: workspaces/src/network/mod.rs (hunk 2 above)
Comment (ChaoticTempest) [reply workspaces/src/network/mod.rs:160]: yeah, was binding to a variable to point out to readers that the resp object is empty, but it's not a big deal, so I'll just get rid of it

**Review by austinabell — APPROVED** [review]
Didn't notice that you moved the logic similar to #102 for removing StatePatcher. I'd say pull this one in first and then I can resolve conflicts on mine

**Comment by ChaoticTempest** [comment]
sounds good then

**Pull request merged**